controllers/remote/cluster_cache.go

```diff
@@ -161,16 +164,28 @@ func (t *ClusterCacheTracker) newClusterAccessor(ctx context.Context, cluster cl
 		return nil, errors.Wrapf(err, "error fetching REST client config for remote cluster %q", cluster.String())
 	}
 
-	// Create a mapper for it
-	mapper, err := apiutil.NewDynamicRESTMapper(config)
+	c, mapper, err := t.createClient(config, cluster)
 	if err != nil {
-		return nil, errors.Wrapf(err, "error creating dynamic rest mapper for remote cluster %q", cluster.String())
+		return nil, err
 	}
 
-	// Create the client for the remote cluster
-	c, err := client.New(config, client.Options{Scheme: t.scheme, Mapper: mapper})
+	runningOnCluster, err := runningOnCluster(ctx, c, cluster)
```

Comment:
I assume this call would go through because of the current hairpinned host on Azure?

Reply:
If I got it correctly, the CAPZ issue is not that it can't communicate at all with the workload cluster. The problem is rather that the connection is only flaky (even when running on control plane nodes).

So I would expect the following:
* if the call works the first time, we detect ourselves in the workload cluster and use the in cluster config from then on (+ it's cached)
* if it doesn't work (all errors != not found), we would get an error here, which would fail the entire reconcile. We now enter a exponential backoff and at some point we should be able to get that call through

I think it doesn't make the situation worse.

In cases where there is no connectivity to the workload cluster at all it would just entirely fail (as of today)
In cases where we have a flaky connection, it will detect self-hosted on the first successful call and then use in-cluster from then on.


IIRC we already need some connectivity to the workload cluster because the `client.New` call already communicates with the apiserver (not 100% sure if I remember this correctly, but it doesn't really matter)

Reply:
@yastij @josh-ferrell @CecileRobertMichon to confirm above assumptions (it seems correct to me, but let's double check)